skimage/morphology/_util.py

```diff
@@ -51,35 +54,182 @@ def _validate_connectivity(image_dim, connectivity, offset):
     return c_connectivity, offset
 
 
-def _offsets_to_raveled_neighbors(image_shape, structure, center, order='C'):
+def _offsets_to_raveled_neighbors(image_shape, selem, center, order='C'):
     """Compute offsets to a samples neighbors if the image would be raveled.
 
     Parameters
     ----------
-    image_shape : tuple
+    image_shape : sequence
         The shape of the image for which the offsets are computed.
-    structure : ndarray
+    selem : ndarray
         A structuring element determining the neighborhood expressed as an
         n-D array of 1's and 0's.
     center : sequence
-        Tuple of indices specifying the center of `selem`.
+        Indices specifying the center of `selem`.
+    order : {"C", "F"}, optional
+        Whether the image described by `image_shape` is in row-major (C-style)
+        or column-major (Fortran-style) order.
 
     Returns
     -------
-    offsets : ndarray
+    raveled_offsets : ndarray
         Linear offsets to a samples neighbors in the raveled image, sorted by
-        their Euclidean distance from the center.
+        their distance from the center.
+
+    Notes
+    -----
+    This function will return values even if `image_shape` contains a dimension
+    length that is smaller than `selem`.
 
     Examples
     --------
     >>> _offsets_to_raveled_neighbors((4, 5), np.ones((4, 3)), (1, 1))
     array([-5, -1,  1,  5, -6, -4,  4,  6, 10,  9, 11])
     """
-    structure = structure.copy()  # Don't modify original input
-    structure[tuple(center)] = 0  # Ignore the center; it's not a neighbor
-    connection_indices = np.transpose(np.nonzero(structure))
-    offsets = (np.ravel_multi_index(connection_indices.T, image_shape,
-                                    order=order) -
-               np.ravel_multi_index(center, image_shape, order=order))
-    squared_distances = np.sum((connection_indices - center) ** 2, axis=1)
-    return offsets[np.argsort(squared_distances)]
+    selem_indices = np.array(np.nonzero(selem)).T
+    offsets = selem_indices - center
+
+    if order == 'F':
+        offsets = offsets[:, ::-1]
+        image_shape = image_shape[::-1]
+    elif not order == 'C':
+        raise ValueError("order was not 'C' or 'F'")
+
+    # Scale offsets in each dimension and sum
+    ravel_factors = image_shape[1:] + (1,)
```

Comment:
if `image_shape` is allowed to be any sequence (as you modified it in the doc), it has to be casted to `tuple` for this line to be correct.

Reply:
Good catch! I reverted that change. Giving the shape as anything but a tuple is non-standard anyway.